testing/ldc/APKBUILD

```diff
@@ -0,0 +1,110 @@
+# -*- shell-script -*-
+# Contributor: Mathias LANG <pro.mathias.lang@gmail.com>
+# Maintainer: Mathias LANG <pro.mathias.lang@gmail.com>
+#
+# This package is divided between the actual package,
+# and the two subpackages, `static` and `runtime`.
+# The main package depends on them.
+# A user might want to install `runtime` when they have
+# a dynamically-linked D program.
+pkgname="ldc"
+pkgver="1.8.0"
+pkgrel=0
+pkgdesc="The LLVM-based D Compiler"
+url="https://github.com/ldc-developers/ldc"
+arch="x86_64"
+license="BSD-3-Clause; Some parts BSL-1.0; Some parts Artistic-1.0 or GPL-2.0-or-later; Some parts NCSA; Some parts NCSA and MIT"
+depends="libexecinfo"
+makedepends="chrpath cmake llvm5-libs libexecinfo-static"
+install=""
+subpackages="$pkgname-runtime $pkgname-doc $pkgname-static"
+provides="dlang-compiler"
+source="https://github.com/ldc-developers/ldc/releases/download/v1.8.0/ldc2-1.8.0-alpine-linux-x86_64.tar.xz
+		ldc2.package.conf"
+builddir="$srcdir/"
+
+build() {
+	rootdir="${srcdir}/ldc2-1.8.0-alpine-linux-x86_64/"
+
+	# Unfortunately the upstream-provided version has the builder's /home/ in its rpath - strip it
+	chrpath -d "${rootdir}/bin/ldc2"
+	chrpath -d "${rootdir}/bin/ldmd2"
+	# This exe is not installed, just used in build, but do it for completeness
+	chrpath -d "${rootdir}/bin/ldc-build-runtime"
+
+	# Now build the runtime - the compiler itself is already built by upstream
+	"${rootdir}"/bin/ldc-build-runtime --buildDir="${builddir}/ldc-runtime-build/"
+
+	# CMake added the rpaths to the shared libs - strip them
+	chrpath -d "${builddir}"/ldc-runtime-build/lib/*.so
+	:
+}
+
+check() {
+	rootdir="${srcdir}/ldc2-1.8.0-alpine-linux-x86_64/"
+	libroot="${builddir}/ldc-runtime-build/lib/"
+
+	echo 'import std.stdio; void main () { try throw new Exception(null); catch (Exception) writeln("Hello World!"); }' > hello_world.d
+	if [ "$(${rootdir}/bin/ldmd2 -conf='' -I${rootdir}/import/ldc -I${rootdir}/import/ -L-L${libroot} -defaultlib=phobos2-ldc,druntime-ldc -L-lexecinfo -L-rpath=${libroot} -run hello_world.d)" != "Hello World!" ]; then
+		return 1
+	fi
+	:
```

Comment:
I'm pretty sure we have a policy against making trivial test cases like these since they can make a package to appear to be working when it actually doesn't

Reply:
What would be a proper test case ?
This did help me a lot during development, as I could ensure programs are linked correctly (D uses `backtrace` which is in Glibc but not in Musl).
When we're building from source we could run the compiler test suite, although it is quite long. I couldn't find clear guidelines on the wiki on what is a good `check` function, w.r.t time, complexity, space, is there such a description somewhere ?

Reply:
A good check function is what upstream provides, so once we have the proper compiler we should use the testsuite you've mentioned. The length doesn't matter, we have plenty of build time to ensure our packages work :)